pkg/apis/applicationnetworking/v1alpha1/vpcassociationpolicy_types.go

```diff
@@ -0,0 +1,70 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/gateway-api/apis/v1alpha2"
+
+	"github.com/aws/aws-application-networking-k8s/pkg/k8s"
+)
+
+const (
+	VpcAssociationPolicyKind = "VpcAssociationPolicy"
+)
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:categories=gateway-api,shortName=vap
+// +kubebuilder:storageversion
+// +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+type VpcAssociationPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec VpcAssociationPolicySpec `json:"spec"`
+}
+
+// +kubebuilder:object:root=true
+// VpcAssociationPolicyList contains a list of VpcAssociationPolicies.
+type VpcAssociationPolicyList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []VpcAssociationPolicy `json:"items"`
+}
+
+// +kubebuilder:validation:MaxLength=32
+// +kubebuilder:validation:MinLength=3
+// +kubebuilder:validation:Pattern=`^sg-[0-9a-z]+$`
+type SecurityGroupId string
+
+// VpcAssociationPolicySpec defines the desired state of VpcAssociationPolicy.
+type VpcAssociationPolicySpec struct {
+
+	// SecurityGroupIds defines the security groups enforced on the VpcServiceNetworkAssociation.
+	// For more details, please check the VPC Lattice documentation https://docs.aws.amazon.com/vpc-lattice/latest/ug/security-groups.html
+	//
+	// +optional
+	// +kubebuilder:validation:MinItems=1
+	SecurityGroupIds []SecurityGroupId `json:"securityGroupIds,omitempty"`
+
+	// AssociateWithVpc indicates whether the VpcServiceNetworkAssociation should be created for current the VPC of EKS cluster.
+	//
+	// Both this flag and Gateway annotation "application-networking.k8s.aws/lattice-vpc-association" are reserved tentatively for backward compatibility.
+	// Either one of them set to true or both of them undefined will result in the VpcServiceNetworkAssociation created.
+	// +optional
+	// +kubebuilder:default=true
+	AssociateWithVpc *bool `json:"associateWithVpc,omitempty"`
```

Comment:
looking at this I'm thinking it is quite not straightforward to have optional value true by default. Could we have an inverse of this option?

Reply:
I can remove  the `// +kubebuilder:default=true` here


I will do the in the controller code logic: ` Either one of them set to true or both of them undefined will result in the VpcServiceNetworkAssociation created.`